### PR TITLE
[WIP] Deck sharing

### PR DESCRIPTION
**Pull request by wrmacrae**
#115 
Do not merge, this is still a work in progress. But getting a PR up in case there are thoughts on the approach so far (notably, adding deck info into logs provides nice psuedo-persistence, but also means sending logs out to players repeatedly). And the UI could show shared decks in more/fewer/different places in theory.

TODO:
- [x] Add deck info to logs correctly
- [x] Log view doesn't show Deck in dropdown if no data
- [x] Show basic lands in logs
- [x] Show sideboard in logs
- [x] Replace existing logs when new one is shared
- [x] Add decks to brackets
- [x] No sharing into log in sealed
- [x] sendLogs handles delayed correctly

Nice to haves:
- [x] Handle repeat sharing by same player (update deck but show timestamp, consider an option to disable this)

### CODE REVIEW

**Comment by tooomm** [comment]
Did you thought about a new option to send logs/decks only to your team members?

**Comment by Senryoku** [comment]
I've been thinking about this feature for a while, thanks for starting the PR. I think storing decks in the draft log is the right approach, we can probably get around re-sending the entire log by using the session id + timestamp as an id to only update the decks.

Another feature that's been requested and require deck sharing is deck hashes. Cockatrice and Magic Workstation uses hashes to verify your opponent's deck, I started working on it until I realized they used the deck (with sideboard and basic) rather than the naked card list. The best source for the hashes I found is their implementations in dr4aft (https://github.com/dr4fters/dr4ft/blob/master/backend/hash.js). This also changes the behaviour of "delayed" draft logs imho (I'd like to find a better name, "locked" doesn't feel right either): They should be sent to everyone but displays only the hashes of shared decks until the "lock" flag is cleared by the owner.

I also like the suggestion (was it by @tooomm ?) of sharing deck on export, rather than adding another button. Not sure about pros and cons there.

**Comment by Senryoku** [comment]
I think this is in a stable state, but some testing/feedback would be appreciated.
Important changes:
- The last received draft log is considered to be "current", meaning it will be used by the bracket to display decks. It's also used to infer that a draft occurred and we can submit our deck.
- Decks can be submitted after a draft (i.e. when there's a valid "current" draft log): They are added to the draft log and broadcast to every player.
- Once submitted, decks are visible in the draft log and in the bracket by clicking on the corresponding player name (discoverability can probably be improved here).
- "Delayed" draft logs are now sent to everyone (a stripped version) so every player can consult submitted deck hashes. The full version is still only sent to the session owner and unlocked by the session owner.

**Comment by wrmacrae** [comment]
Will read through the changes more closely from a tech perspective, but from testing, things mostly look good. A few ideas for improvement (all of them can wait till post-merge and happen separately):
- Keep the columns set up during brewing in the shared deck (vs reverting to CMC columns)
- Add statistics button in shared deck view
- The export buttons in draft log are a little confusing now (which ones will export the deck vs the log)
- I couldn't test the /bracket view on the review deployment, so not sure how that looks/works at the moment.

Thanks for picking this up - excited to get it out to people, especially for future sets before their Arena release!

File: client/src/components/Bracket.vue [hunk 1]
```diff
@@ -101,7 +121,7 @@ export default {
 				return this.records[p].wins === 3;
 			}
 		},
-		isSilver: function (p, index) {
+		isSilver: function (p /*, index*/) {
```
Comment (wrmacrae) [inline client/src/components/Bracket.vue:124]: remove dead code

**Comment by wrmacrae** [comment]
I think we can probably merge this and I can file the more minor improvements in new issues and tackle those separately. In testing it seemed like the read-only brackets lost state sometimes, so I'm a little curious to see how well this work in production with real players logged, but I think it's best to get things out there so we can see usage and collect feedback

**Comment by tooomm** [comment]
As I didn't manage to check this out yet, I'll add to your new issue then @wrmacrae

**Pull request merged**